scripts/zones/Halvung/npcs/_1qp.lua

```diff
@@ -0,0 +1,34 @@
+-----------------------------------
+-- Area: Halvung
+--  NPC: Decorative Bronze Gate (_1qp)
+-----------------------------------
+local ID = require("scripts/zones/Halvung/IDs")
+require("scripts/globals/npc_util")
+require("scripts/globals/items")
+require("scripts/globals/status")
+-----------------------------------
+local entity = {}
+
+entity.onTrade = function(player, npc, trade)
+    if npcUtil.tradeHasExactly(trade, {2221, 2222, 2223}) then
+        player:confirmTrade()
+        npc:openDoor()
+        player:messageSpecial(ID.text.KEY_BREAKS, 2221,2222,2223)
```

Comment:
still need these magic numbers out.
```lua
entity.onTrade = function(player, npc, trade)
    if
        npcUtil.tradeHasExactly(trade,
        {
            xi.items.HALVUNG_SHAKUDO_KEY,
            xi.items.HALVUNG_BRONZE_KEY,
            xi.items.HALVUNG_BRASS_KEY
        })
    then
        player:confirmTrade()
        npc:openDoor()
        player:messageSpecial(ID.text.KEY_BREAKS, 
            xi.items.HALVUNG_SHAKUDO_KEY,
            xi.items.HALVUNG_BRONZE_KEY,
            xi.items.HALVUNG_BRASS_KEY
        )
    end
end
```
Unsure of proper formatting of these widelines, this is my best guess.

@claywar @zach2good this doesn't exactly fit with examples in our style guide, I think. Care to chime in?

Reply:
Yes, I am working literally on that right now ;) Testing